### PR TITLE
Update error message returned when user is not allowed to sign db certs

### LINKED ISSUE #12455: Provide more information on access denied for tctl auth sign
## What would you like Teleport to do?

Provide information on why a user's action of `tctl auth sign` is denied.  In this case a user doesn't have the impersonate right in their role.

## What problem does this solve?

Users just receive this obtuse error that doesn't give any indication on how to resolve.

image

## Workaround
Check back in guide for role information.

### PR DESCRIPTION
**Pull request by GavinFrazar**
Closes #12455

This just updates the error message a user will see with `tctl auth sign --format=db` or any other database format. I updated the docs page that the error messages linked to as well, to include a little more info about the requirement for impersonation if tctl is not run on the auth server.

### CODE REVIEW

**Review by ryanclark — APPROVED** [review]

File: docs/pages/database-access/reference/cli.mdx [hunk 1]
```diff
@@ -136,7 +136,9 @@ $ teleport db configure bootstrap -c /etc/teleport.yaml --manual
 
 When invoked with a `--format=db` (or `--format=mongodb` for MongoDB) flag,
 produces a CA certificate, a client certificate and a private key file used for
-configuring Database Access with self-hosted database instances.
+configuring Database Access with self-hosted database instances. Note that for database
+formats, tctl must be run on the auth server or the remote
+user must be be able to impersonate the database service.
```
Comment (nklaassen) [inline docs/pages/database-access/reference/cli.mdx:141]: I'd consider putting this in an `<Admonition type="note">`

Also "the remote user must be be able to impersonate the database service" doesn't help me much, even as a teleport dev. I think our docs on impersonating services are basically non-existent, but you could at least say "must be able to impersonate the builtin role `Db`, see the impersonation guide (link) for details on how to allow impersonation" (I'm not sure if that role name is right or if they need a user as well)

File: lib/auth/auth_with_roles.go [hunk 2]
```diff
@@ -3610,7 +3610,7 @@ func (a *ServerWithRoles) GenerateDatabaseCert(ctx context.Context, req *proto.D
 		if err := a.canImpersonateBuiltinRole(types.RoleDatabase); err != nil {
 			log.WithError(err).Warnf("User %v tried to generate database certificate but is not allowed to impersonate %q system role.",
 				a.context.User.GetName(), types.RoleDatabase)
-			return nil, trace.AccessDenied("access denied")
+			return nil, trace.AccessDenied("access denied. This user is not authorized to create certificates, follow the following guide https://goteleport.com/docs/database-access/reference/cli/#tctl-auth-sign.")
```
Comment (nklaassen) [inline lib/auth/auth_with_roles.go:3613]: We know the exact problem here, would be nice to just share it imo
```suggestion
			return nil, trace.AccessDenied(`access denied. The user must be able to impersonate the role "Db" in order to generate database certificates, for more info see https://goteleport.com/docs/database-access/reference/cli/#tctl-auth-sign.`)
```

File: docs/pages/database-access/reference/cli.mdx (hunk 1 above)
Comment (GavinFrazar) [reply docs/pages/database-access/reference/cli.mdx]: Added

**Review by smallinsky — APPROVED** [review]

**Review by nklaassen — APPROVED** [review]

File: docs/pages/database-access/reference/cli.mdx [hunk 3]
```diff
@@ -139,6 +139,12 @@ When invoked with a `--format=db` (or `--format=mongodb` for MongoDB) flag,
 produces a CA certificate, a client certificate and a private key file used for
 configuring Database Access with self-hosted database instances.
 
+<Admonition type="note" title="Note">
+  For database formats, tctl must be run on the auth server or the remote
```
Comment (ptgott) [inline docs/pages/database-access/reference/cli.mdx:143]:
```suggestion
  For database formats, `tctl` must be run on an Auth Service host or the remote
```
To align with the rest of the docs

File: docs/pages/database-access/reference/cli.mdx [hunk 4]
```diff
@@ -139,6 +139,12 @@ When invoked with a `--format=db` (or `--format=mongodb` for MongoDB) flag,
...
+  user must be be able to impersonate the builtin role and user "Db". See the [impersonation guide](../../access-controls/guides/impersonation.mdx)
```
Comment (ptgott) [inline docs/pages/database-access/reference/cli.mdx:144]:
```suggestion
  user must be be able to impersonate the built-in `Db` role and user. See the [impersonation guide](../../access-controls/guides/impersonation.mdx)
```

**Review by ptgott — APPROVED** [review]
Approved with minor comments. Disabled auto-merge so you can incorporate them.

**Comment by GavinFrazar** [comment]
> Approved with minor comments. Disabled auto-merge so you can incorporate them.

Thanks!

**Review by ptgott — APPROVED** [review]

**Comment by smallinsky** [comment]
@GavinFrazar Can we merge this ?

**Pull request merged**